### PR TITLE
add partner support to sdk-unify

### PR DESCRIPTION
**Pull request by barchen1**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/24385
fixes: https://github.com/demisto/etc/issues/19984
fixes: https://github.com/demisto/etc/issues/22552

## Description
Added to unify command support to contributed packs (partner, developer, community) and now it adds contributor details to the display name and description.

## Screenshots
Screen Shot 2020-06-10 at 17 34 32

Screen Shot 2020-06-10 at 17 33 07

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: demisto_sdk/commands/unify/unifier.py [hunk 1]
```diff
@@ -338,3 +349,44 @@ def clean_pwsh_code(script_code):
...
+            partner_url (str): The partner url.
+
+        Returns:
+            The unified yaml file.
+        """
+        unified_yml['display'] += PARTNER_CONTRIB_DISPLAY_NAME
+        existing_detailed_description = unified_yml['detaileddescription']
+        partner_description = PARTNER_CONTRIB_DETAILED_DESC
+        if partner_email:
+            partner_description += f'\n**Email** - {partner_email}'
```
Comment (ronykoz) [inline demisto_sdk/commands/unify/unifier.py:387]:
```suggestion
            partner_description += f'\n**Email** - [mailto](mailto:{partner_email})'
```

File: demisto_sdk/commands/unify/unifier.py [hunk 2]
```diff
@@ -338,3 +349,44 @@ def clean_pwsh_code(script_code):
...
+        if partner_url:
+            partner_description += f'\n**URL** - {partner_url}'
```
Comment (ronykoz) [inline demisto_sdk/commands/unify/unifier.py:389]:
```suggestion
            partner_description += f'\n**URL** - [{partner_url}]({partner_url})'
```

File: demisto_sdk/commands/unify/tests/unifier_test.py [hunk 3]
```diff
@@ -501,3 +488,112 @@ def test_unify_default_output_script(self):
...
+PARTNER_DISPLAY_NAME = 'test (Partner contribution)'
+PARTNER_DETAILEDDESCRIPTION = '### This is a Partner Contributed integration' \
+    f'\nFor all questions and enhancement requests please contact the partner directly:' \
+                              f'\n**Email** - {PARTNER_EMAIL}\n**URL** - {PARTNER_URL}\n***\ntest details'
+
+
+def test_unify_partner_contributed_pack(mocker, repo):
+    """
+    Given
+        - Partner contributed pack
```
Comment (ronykoz) [inline demisto_sdk/commands/unify/tests/unifier_test.py:549]: with email and url in the support details

File: demisto_sdk/commands/unify/tests/unifier_test.py [hunk 4]
```diff
@@ -501,3 +488,112 @@ def test_unify_default_output_script(self):
...
+    assert 'Created unified yml:' in result.stdout
+    # Verifying the unified file data
+    assert PARTNER_UNIFY["display"] == PARTNER_DISPLAY_NAME
+    assert PARTNER_UNIFY["detaileddescription"] == PARTNER_DETAILEDDESCRIPTION
+
+
+def test_unify_not_partner_contributed_pack(mocker, repo):
+    """
+    Given
+        - Normal pack - not a partner contribution
```
Comment (ronykoz) [inline demisto_sdk/commands/unify/tests/unifier_test.py:577]:
```suggestion
        - XSOAR supported pack - not a partner contribution
```

File: demisto_sdk/commands/unify/tests/unifier_test.py [hunk 5]
```diff
@@ -501,3 +488,112 @@ def test_unify_default_output_script(self):
...
+}
+NORMAL_UNIFY = PARTNER_UNIFY.copy()
+INTEGRATION_YAML = {'display': 'test', 'script': {'type': 'python'}}
+
+PARTNER_DISPLAY_NAME = 'test (Partner contribution)'
+PARTNER_DETAILEDDESCRIPTION = '### This is a Partner Contributed integration' \
+    f'\nFor all questions and enhancement requests please contact the partner directly:' \
+                              f'\n**Email** - {PARTNER_EMAIL}\n**URL** - {PARTNER_URL}\n***\ntest details'
+
+
```
Comment (ronykoz) [inline demisto_sdk/commands/unify/tests/unifier_test.py:545]: add a test with only url and then only email

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 6]
```diff
@@ -189,6 +193,21 @@ def _is_pack_meta_file_structure_valid(self):
...
+        """Checks if email or url exist in contributed pack details."""
+        try:
+            pack_meta_file_content = json.loads(self._read_file_content(self.pack_meta_file))
+            if pack_meta_file_content[PACK_METADATA_SUPPORT] in CONTRIBUTORS_LIST:
+                if not pack_meta_file_content[PACK_METADATA_URL] and not pack_meta_file_content[PACK_METADATA_EMAIL]:
+                    if self._add_error(Errors.pack_metadata_missing_url_and_email(), self.pack_meta_file):
+                        return False
+
+        except (ValueError, TypeError):
+            if self._add_error(Errors.pack_metadata_isnt_json(self.pack_meta_file), self.pack_meta_file):
```
Comment (reutshal) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:206]: Maybe save `Errors.pack_metadata_isnt_json(self.pack_meta_file), self.pack_meta_file` as a variable at the beggining of the method

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py [hunk 7]
```diff
@@ -1,9 +1,32 @@
+import json
 import os
```
Comment (reutshal) [inline demisto_sdk/commands/common/tests/pack_unique_files_test.py:2]:
```suggestion
import os
import json
```

File: demisto_sdk/commands/unify/unifier.py [hunk 8]
```diff
@@ -338,3 +351,46 @@ def clean_pwsh_code(script_code):
         script_code = script_code.replace(". $PSScriptRoot\\demistomock.ps1", "")
         script_code = script_code.replace(". $PSScriptRoot\\CommonServerPowerShell.ps1", "")
         return script_code
+
+    def get_pack_path(self):
+        return self.package_path.split('Integrations')[0]
+
+    def is_contributor_pack(self):
```
Comment (reutshal) [inline demisto_sdk/commands/unify/unifier.py:358]: The name does not really describe this method.
I would suggest something like `get_contributor_data` or something of that sort.
I would expect this name to return an only boolean result.
This change will also affect the docstring description.

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py (hunk 6 above)
Comment (ronykoz) [reply demisto_sdk/commands/common/hook_validations/pack_unique_files.py:206]: no need, this is the way this whole file is written

File: demisto_sdk/commands/common/tests/pack_unique_files_test.py (hunk 7 above)
Comment (ronykoz) [reply demisto_sdk/commands/common/tests/pack_unique_files_test.py:2]: we have isort, we can't touch this

File: demisto_sdk/commands/unify/unifier.py (hunk 8 above)
Comment (ronykoz) [reply demisto_sdk/commands/unify/unifier.py:358]: i agree, should have one for the boolean, and the other one for the data.

the data one should be called from the add contributors data

**Pull request merged**